src/proposal.mligo

```diff
@@ -73,6 +79,45 @@ let is_ge_qt(qt_1, qt_2 : quorum_threshold * quorum_threshold): bool =
   let (nat_1, nat_2) = cmp_qt (qt_1, qt_2) in
   nat_1 >= nat_2
 
+[@inline]
+// Multiply two fractions
+let mul_qt(qt_1, qt_2 : quorum_threshold * quorum_threshold): quorum_threshold =
+  { numerator = qt_1.numerator * qt_2.numerator; denominator = qt_1.denominator * qt_2.denominator }
+
+// Divide the first fraction by the second
+let div_qt(qt_1, qt_2 : quorum_threshold * quorum_threshold): quorum_threshold =
+  { numerator = qt_1.numerator * qt_2.denominator
+  ; denominator = qt_1.denominator * qt_2.numerator
+  }
+
+[@inline]
+// Adds two fractions
+let add_qt(qt_1, qt_2 : quorum_threshold * quorum_threshold): quorum_threshold =
+  { numerator = (qt_1.numerator * qt_2.denominator) + (qt_2.numerator * qt_1.denominator); denominator = qt_1.denominator * qt_2.denominator }
+
+[@inline]
+// Subtracts two fractions, and throws error if second one is larger.
+let sub_qt(qt_1, qt_2 : quorum_threshold * quorum_threshold): quorum_threshold = let
+  n = match is_nat((qt_1.numerator * qt_2.denominator) - (qt_2.numerator * qt_1.denominator)) with
+    | Some n -> n
+    | None -> ((failwith "UNEXPECTED_NEGATIGE_FRACTION_RESULT") : nat) in
+  { numerator = n; denominator = qt_1.denominator * qt_2.denominator }
+
+[@inline]
+let rec round(qt: quorum_threshold): quorum_threshold =
+  if (qt.denominator > 10000n && qt.numerator > 1000n)
+    then round({numerator = qt.numerator / 100n; denominator = qt.denominator / 100n })
+    else qt
+
+let bound_qt (qt, min_qt, max_qt : quorum_threshold * quorum_threshold * quorum_threshold)
+    : quorum_threshold =
+  if is_gt_qt(qt, max_qt)
+  then max_qt
+  else
+    if is_le_qt(qt, min_qt)
+    then min_qt
+    else qt
```

Comment:
Now that we use this here and more than once it makes sense to inline it.

Also I think it can be redefined to only use `is_gt_qt`, which I suggest to do to have one less function in the code.
```suggestion
[@inline]
let bound_qt (qt, min_qt, max_qt : quorum_threshold * quorum_threshold * quorum_threshold)
    : quorum_threshold =
  if is_gt_qt(qt, max_qt)
  then max_qt
  else if is_gt_qt(min_qt, qt)
  then min_qt
  else qt
```